ouroboros-consensus/src/Ouroboros/Consensus/Node.hs

```diff
@@ -381,12 +381,13 @@ newtype LastShutDownWasClean = LastShutDownWasClean Bool
 --
 -- Run the body action with the DB locked, and if the last shutdown was clean.
 --
-stdWithCheckedDB :: forall a.
-     FilePath
+stdWithCheckedDB :: forall blk a. (StandardHash blk, Typeable blk) =>
+     Proxy blk
+  -> FilePath
   -> NetworkMagic
   -> (LastShutDownWasClean -> IO a)  -- ^ Body action with last shutdown was clean.
   -> IO a
-stdWithCheckedDB databasePath networkMagic body = do
+stdWithCheckedDB blk databasePath networkMagic body = do
```

Comment:
We have a convention to use `pb` (or `p`) to name such proxies. `blk` is typically used for actual blocks.